doc/build.txt

```diff
@@ -758,4 +758,34 @@ Older versions in particular are not supported by OpenVDB.
 @b Note: Boost will produce a version error in the format `Detected version of
 Boost is too old.`
 
+@subsection troubleshoot3 CMake warnings/errors in FindBoost.cmake
+
+There are a variety of potential warnings and errors that may arise from the
+version compatibility of Boost and CMake. These are related to how both Boost
+and CMake have historically been kept in sync. A common warning and error
+combination is:
+
+@code{.sh}
+  CMake Warning at ...  Imported targets not available for Boost version XXXXXX
+  ...
+  Target "..." links to target "Boost::XXX" but the target
+  was not found.  Perhaps a find_package() call is missing for an IMPORTED
+  target, or an ALIAS target is missing?
+@endcode
+
+OpenVDB uses [imported targets](https://cmake.org/cmake/help/latest/command/add_library.html#imported-libraries)
+for all its dependencies. For imported Boost compatibility, the following
+versions of CMake are required:
+
+  - Boost 1.63 requires CMake 3.7 or newer.
+  - Boost 1.64 requires CMake 3.8 or newer.
+  - Boost 1.65 and 1.65.1 require CMake 3.9.3 or newer.
+  - Boost 1.66 requires CMake 3.11 or newer.
+  - Boost 1.67 requires CMake 3.12 or newer.
+  - Boost 1.68, 1.69 require CMake 3.13 or newer.
+  - Boost 1.70 requires CMake 3.14 or newer.
```

Comment:
The Boost/CMake relationship is a mess. It sounds like they are attempting to review this post 1.70 and 1.70 does come with a different CMake implementation (I think). I've chosen to add this comment here rather than implement this logic in our CMake as I'm not sure how strict we should be.

An interesting point is that if we do want to continue to use imported targets, our minimum CMake version will jump to 3.11 for VFX18. Worth discussion now.

Reply:
Yes, it was more whether or not 3.11 is the correct version to bump to i.e why not higher?

Reply:
I think the question is why not lower?

Right now it feels like CMake 3.11 is a good balance between picking up the more recent features of CMake and not being too bleeding edge so as to minimize the chances that users won't need to install a newer version that's already available to them. It also nicely ties in with Boost 1.66 which is the lowest supported version for VFX 2019. If there's a good reason to go higher than 3.11 to get access to a particular feature, that's worth talking about though.

Reply:
> I think the question is why not lower?

Because of the boost version (1.66) requiring 3.11 for imported targets.

3.12 would allow us to remove some of the python branching logic AND (I think) brings native support for `XXX_ROOT` variables which would be incredibly useful. 3.14 would remove all conditional branching logic. So as 3.14 seems a bit too much, the choice is between 3.11 and 3.12.